### PR TITLE
51383 detail course with list subject

### PR DESCRIPTION
**Pull request by vietbq**

### CODE REVIEW

File: app/views/users/_courses_options.html.erb [hunk 1]
```diff
@@ -0,0 +1,17 @@
+<div class="col-md-5 col-sm-5 col-xs-6">
+  <%= label_tag t("course.title"), nil%>
+  <%= select_tag "course", options_for_select(@user.user_courses.collect{
+    |user_course| [user_course.course.name, user_course.id]}),
+    class: "form-control select_tag" %>
+</div>
+<div class="col-md-5 col-sm-5 col-xs-6">
+  <%= label_tag t("course.start"), nil, class: "small_label" %>
+  <span class="timestamp ">
```
Comment (hellovietnam93) [inline app/views/users/_courses_options.html.erb]: remove space before ```"```

File: app/views/users/_courses_options.html.erb [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+    <%= @user_course.course.start_date.strftime("%d/%m/%Y") +
+      " - " + @user_course.course.end_date.strftime("%d/%m/%Y")%>
+  </span>
+</div>
+<div class="col-md-2 col-sm-2">
+  <% get_status @user_course.status %>
+  <%= button_tag @check_box_text, class: "btn " + @task_class %>
```
Comment (hellovietnam93) [inline app/views/users/_courses_options.html.erb]:
```"btn #{@task_class}"

File: app/views/users/_list_subjects.html.erb [hunk 3]
```diff
@@ -0,0 +1,36 @@
+<% @user.user_courses.first.course.course_subjects.each do |course_subject| %>
```
Comment (hellovietnam93) [inline app/views/users/_list_subjects.html.erb]: set a variable in controller.
it is too long

File: app/views/users/_list_subjects.html.erb [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+  <% get_subject_status course_subject.id %>
+  <% subject_class = "btn subject-button-width " + @task_class %>
```
Comment (hellovietnam93) [inline app/views/users/_list_subjects.html.erb]: ```"btn subject-button-width #{@task_class}" ```

File: app/views/users/_list_subjects.html.erb [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+          <th><%= t "subject.start" %>
+          <span class="timestamp" style="color: red">
+            <%= course_subject.course.start_date.strftime("%d/%m/%Y") +
+              " - " + course_subject.course.end_date.strftime("%d/%m/%Y")%>
+          </span>
+          </th>
+          <th><%= button_tag @check_box_text, class: subject_class %></th>
+          <th>
+            <%= button_tag t("course.hide"), type: "button",
+              onclick: "hidden_content('#task-in-profle-" + course_subject.id.to_s + "')",
```
Comment (hellovietnam93) [inline app/views/users/_list_subjects.html.erb]: don't use onclick in view, find a other solution

File: app/views/users/_list_subjects.html.erb [hunk 6]
```diff
@@ -0,0 +1,36 @@
...
+              class: 'btn btn-default' %>
```
Comment (hellovietnam93) [inline app/views/users/_list_subjects.html.erb]: "" instead of ''

File: app/views/users/_profile.html.erb [hunk 7]
```diff
@@ -0,0 +1,17 @@
+<div class="col-md-3" align="center">
+  <%= gravatar_for @user, options = {size: 200} %>
```
Comment (hellovietnam93) [inline app/views/users/_profile.html.erb]: Settings

File: app/views/users/show.html.erb [hunk 8]
```diff
@@ -4,21 +4,18 @@
...
-              <td><%= @user.email %></td>
-            </tr>
-          </tbody>
-        </table>
-      </div> 
+      <%= render "profile" %>
+    </div>
+    <div class="row">
+      <div class="col-md-8 col-sm-12  col-xs-12">
+        <div class="col-md-12 col-sm-12  col-xs-12">
```
Comment (hellovietnam93) [inline app/views/users/show.html.erb]: remove space

File: app/views/users/_list_subjects.html.erb [hunk 9]
```diff
@@ -0,0 +1,36 @@
...
+  <% get_subject_status course_subject.id %>
+  <% subject_class = "btn subject-button-width " + @task_class %>
+  <div class="col-md-12">
+    <table class="table"> 
+      <thead> 
+        <tr class="active"> 
+          <th><b><center><%= course_subject.subject.name %></center></b></th>
+          <th><%= t "subject.start" %>
+          <span class="timestamp" style="color: red">
+            <%= course_subject.course.start_date.strftime("%d/%m/%Y") +
```
Comment (hellovietnam93) [inline app/views/users/_list_subjects.html.erb]: "%d/%m/%Y" set in I18n

File: app/views/users/_list_subjects.html.erb [hunk 10]
```diff
@@ -0,0 +1,38 @@
...
+          </th>
+        </tr>
+      </thead> 
+      <tbody id= <%= "task-in-profle-" + course_subject.id.to_s %> > 
+        <% course_subject.course_subject_tasks.each do |course_subject_task| %>
+          <% user_task_status course_subject_task.id %>
+          <tr> 
+            <th scope="row"><center><%= course_subject_task.task.name %></center></th> 
+            <td></td> 
+            <td><%= button_tag  @check_box_text, class: @task_class %></td>
```
Comment (hellovietnam93) [inline app/views/users/_list_subjects.html.erb]: remove space

File: app/views/users/show.html.erb [hunk 11]
```diff
@@ -4,21 +4,18 @@
...
+    </div>
+    <div class="row">
+      <div class="col-md-8 col-sm-12  col-xs-12">
+        <div class="col-md-12 col-sm-12 col-xs-12">
+          <%= render "courses_options" %>
+          <%= render "list_subjects" %>
+        </div>
+      </div>
+      <div class="col-md-4 col-sx-12">
+          <h3><%= t "course.activity" %></h3>
```
Comment (hellovietnam93) [inline app/views/users/show.html.erb]: ident

**Comment by trikmbk56** [comment]
ok

File: app/assets/stylesheets/custom.css.scss [hunk 12]
```diff
@@ -136,4 +130,28 @@ input {
...
+  color: white;
+}
+.status-finish {
+  background: #9A9A9A;
+  color: white;
+}
+.subject-button-width{
+  width: 85px;
+}
+
```
Comment (hellovietnam93) [inline app/assets/stylesheets/custom.css.scss]: remove this line

File: app/views/users/_courses_options.html.erb [hunk 13]
```diff
@@ -0,0 +1,17 @@
...
+  <%= label_tag t("course.title"), nil%>
+  <%= select_tag "course", options_for_select(@user.user_courses.collect{
+    |user_course| [user_course.course.name, user_course.id]}),
+    class: "form-control select_tag" %>
+</div>
+<div class="col-md-5 col-sm-5 col-xs-6">
+  <%= label_tag t("course.start"), nil, class: "small_label" %>
+  <span class="timestamp">
+    <%= @user_course.course.start_date.strftime(t "time_format") +
+      " - " + @user_course.course.end_date.strftime(t "time_format")%>
```
Comment (hellovietnam93) [inline app/views/users/_courses_options.html.erb]: add space after ```%>```

**Pull request merged**